### PR TITLE
docs: add role and tools to getting started cli experience

### PR DESCRIPTION
**Pull request by psschwei**
Signed-off-by: Paul S. Schweigert <paul@paulschweigert.com>

### Which issue(s) does this pull-request address?

Closes: #148

### Description

We have found that bees perform best when:
* they use fewer number of tools, and
* they are provided explicit instructions/role

Here we updated the getting started section of the README to incorporate those recommendations into the simple example. We update the system prompt to provide explicit instructions, and we remove the search tool as it isn't needed for weather-related queries. It also adds some tips related to these two points to the appropriate section.

Additionally, it adds a yarn helper method to run the simple example: `yarn run quickstart`

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [x] E2E tests pass: `yarn test:e2e`
- [x] Tests are included 
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

no tests included as this is a documentation change

### CODE REVIEW

**Comment by psschwei** [comment]
FYI @mmurad2 - if this wasn't what you were envisioning let me know, happy to iterate on this

**Comment by psschwei** [comment]
this is a new one :smile: 

image

image

File: README.md [hunk 1]
```diff
@@ -74,9 +83,15 @@ const response = await agent
...
+> [!TIP]
+>
+> Providing a bee with specific `instructions` on what their role should be will improve performance and is recommended.
+
+> [!TIP]
+>
+> Bees work best when using a smaller number of `tools`, so it is recommended to only enable the minimum required tools.
 
-➡️ you can run this example after local installation, using the command `yarn start examples/agents/simple.ts`
+➡️ you can run this example after local installation, using the command `yarn run quickstart`
```
Comment (Tomas2D) [inline README.md:94]: `run` is not needed

File: README.md [hunk 2]
```diff
@@ -86,6 +101,8 @@ console.log(`Agent 🤖 : `, response.result.text);
 >
 > Documentation is available at https://i-am-bee.github.io/bee-agent-framework/
 
+➡️ See a more [advanced example](/examples/agents/bee.ts)
```
Comment (Tomas2D) [inline README.md:104]: Missing `.` at the end.

File: README.md [hunk 3]
```diff
@@ -52,15 +52,24 @@ yarn add bee-agent-framework
...
+  tools: [new OpenMeteoTool()], // for more explore 'bee-agent-framework/tools'
+  templates: {
+    system: BeeSystemPrompt.fork((old) => ({
+      ...old,
+      defaults: {
+        instructions:
+          "You are a helpful assistant that uses tools to answer weather-related questions.",
+      },
+    })),
+  },
```
Comment (Tomas2D) [inline README.md:72]: This definitely should not be part of README. Keep the example as is.

**Review by Tomas2D — CHANGES_REQUESTED** [review]
I think the issue was about adding `role` and `tools` to the CLI.
So, when someone starts the script, it first asks for tools and a role, and then the chat begins.

**Comment by psschwei** [comment]
> I think the issue was about adding `role` and `tools` to the CLI.
> So, when someone starts the script, it first asks for tools and a role, and then the chat begins.

Ok, I think I misunderstood the ask here.

Do we currently have a way to create an agent from the command line? I believe all the examples I saw here are running an existing source code file (e.g. `yarn start examples/agents/bee.ts`) which already have tools and instructions predefined. So I'd assume we'd need to edit the source code files to add the tools / role prior to running the file? Or is there another approach that I'm missing?

Would there be a way to use the API here? That seems like the ideal way to add this kind of functionality, but I'm assuming that would need the whole stack and not just the framework, so might not be an option here...

**Comment by Tomas2D** [comment]
Why would you use an API in the framework? API is built on top of the framework.
Just ask the user to specify a role, and if one is not provided then fallback to the default. 

Maybe consult details in the related issue.

**Pull request closed without merging**